pkg/adapters/aiven.go

```diff
@@ -0,0 +1,542 @@
+		case string:
+			settingValue = v
+		case int64:
+			settingValue = v
+		case int:
+			settingValue = int64(v)
+		default:
+			// For other types, convert to string
+			settingValue = fmt.Sprintf("%v", v)
+		}
+
+		adapter.Logger().Infof("Setting %s = %v (service level: %v)", paramName, settingValue, paramInfo.ServiceLevel)
+		if logWriter != nil {
+			fmt.Fprintf(logWriter, "Setting %s = %v (service level: %v)\n", paramName, settingValue, paramInfo.ServiceLevel)
+		}
+
+		// Apply the parameter at the correct level (service level or pg config)
+		if paramInfo.ServiceLevel {
+			userConfig[paramName] = settingValue
+		} else {
+			pgParamsMap[paramName] = settingValue
+		}
+
+		changesMade = true
+	}
+
+	// Only update if changes were made
+	if !changesMade {
+		adapter.Logger().Info("No applicable changes to apply")
+		return nil
+	}
+
+	// Update the user_config with modified pg parameters
+	userConfig["pg"] = pgParamsMap
+
+	// Create the update request
+	updateReq := &service.ServiceUpdateIn{
+		UserConfig: &userConfig,
+		Powered:    boolPtr(true),
+	}
+
+	// Debug: Print the final configuration being sent to Aiven
+	adapter.Logger().Infof("Final update request to be sent to Aiven: %+v", updateReq)
+	if logWriter != nil {
+		fmt.Fprintf(logWriter, "Final update request to be sent to Aiven: %+v\n", updateReq)
+	}
+
+	// Apply the configuration update
+	response, err := adapter.aivenClient.ServiceUpdate(
+		context.Background(),
+		adapter.aivenConfig.ProjectName,
+		adapter.aivenConfig.ServiceName,
+		&service.ServiceUpdateIn{
+			UserConfig: &userConfig,
+			Powered:    boolPtr(true),
+		},
+	)
+
+	adapter.Logger().Infof("Aiven response: %+v", response)
+
+	if err != nil {
+		return fmt.Errorf("failed to update PostgreSQL parameters: %v", err)
+	}
+
+	// Wait for the service to be running again
+	err = adapter.waitForServiceState(service.ServiceStateTypeRunning)
+	if err != nil {
+		return err
+	}
+
+	// Verify PostgreSQL is responding
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timeout waiting for PostgreSQL to come back online")
+		case <-time.After(5 * time.Second):
+			_, err := adapter.pgDriver.Exec(ctx, "SELECT 1")
+			if err == nil {
+				adapter.Logger().Info("PostgreSQL is back online")
+				adapter.state.LastAppliedConfig = time.Now()
+				return nil
+			}
+			adapter.Logger().Debug("PostgreSQL not ready yet, retrying...")
+		}
+	}
+}
+
+// waitForServiceState waits for the service to reach the specified state
+func (adapter *AivenPostgreSQLAdapter) waitForServiceState(state service.ServiceStateType) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
+	defer cancel()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timeout waiting for service to reach state %s", state)
+		case <-time.After(10 * time.Second):
+			service, err := adapter.aivenClient.ServiceGet(
```

Comment:
there is a shadowing of the service name with an imported package, either rename the imported package or this var
```suggestion
			serviceRenameMePlease, err := adapter.aivenClient.ServiceGet(
```